core/src/test/scala/tech/beshu/ror/unit/utils/GraphNodeAncestorsExplorerTest.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.unit.utils
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Positive
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import tech.beshu.ror.utils.GraphNodeAncestorsExplorer
+
+class GraphNodeAncestorsExplorerTest extends AnyWordSpec with Matchers with MockFactory {
+
+  "Parents of 1 are" in {
+    val graph = createDynamicallyTraversableGraph(nestedLevels = 2)
+    graph.findAllAncestorsOf(Set("1")).runSyncUnsafe() should be (Set("2", "3", "4", "6"))
+  }
+
+  "Parents of 2 are" in {
+    val graph = createDynamicallyTraversableGraph(nestedLevels = 1)
+    graph.findAllAncestorsOf(Set("2")).runSyncUnsafe() should be (Set("4"))
+  }
+
+  private def createDynamicallyTraversableGraph(nestedLevels: Int Refined Positive) = {
+    new GraphNodeAncestorsExplorer[String](nestedLevels, doFetchParentNodesOfMock())
+  }
+
+  private def doFetchParentNodesOfMock() = {
+    /**
```

Comment:
Redundant asterisk( I have a rendering of comments enabled and this does not look well :) )
![image](https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/assets/26696870/28376e7d-1946-4a56-b055-8bf9b87df12d)